aten/src/ATen/TensorIterator.cpp

```diff
@@ -905,6 +914,12 @@ TensorIterator TensorIterator::unary_float_op(Tensor& out, const Tensor& a) {
   return iter;
 }
 
+TensorIterator TensorIterator::unary_boolean_op(const Tensor& out, const Tensor& a) {
```

Comment:
One point need to discuss is that, after considering similar operators (like `torch.signbit`) that require boolean casting output, I wrote `unary_boolean_op` for code reuse. I am not sure if that is appropriate or should I build TI directly in meta function?

Reply:
hmm, one reason not to is that it looks like not all "unary boolean" ops follow this same set of config.

E.g. looking at the `logical_not` kernel - it's pretty similar, in that `torch.logical_not(a)` will return a boolean tensor, but it looks like it doesn't enforce that the out= tensor has to have boolean dtype. So you can't unconditionally use `.declare_static_dtype_and_device(at::kBool, a.device())` - you only want to do it when the output is currently not defined (in the functional version of the op).

If you leave this as a TensorIterator API, can you change the name (maybe `unary_force_boolean_op`?), and leave a comment mentioning that this API should only be used when the output tensor must have boolean type?

Reply:
It sounds very reasonable, thanks for such detailed explanation! Let me change the name and leave a comment to describe that.